flink-table-store-core/src/main/java/org/apache/flink/table/store/file/operation/FileStoreReadImpl.java

```diff
@@ -55,14 +56,14 @@ public FileStoreReadImpl(
             WriteMode writeMode,
             RowType keyType,
             RowType valueType,
-            Comparator<RowData> keyComparator,
+            Supplier<Comparator<RowData>> keyComparatorSupplier,
```

Comment:
Why we need to use `Supplier<Comparator<RowData>>`?

Reply:
Since `pickManifest` needs `keyComparator`, so I extract a util method

Reply:
Why a util method will modify `Comparator` to `Supplier`?
We shouldn't create it repeatedly unless there is a thread safety issue here.